docs/upgrading/migrating-local-extensions-33.md

```diff
@@ -0,0 +1,138 @@
+
+    "Bolt\\Extension\\Myname\\MyExtensionNamespace\\": "local/myname/myextension"
+    
+If you don't currently have a psr-4 section in your `extensions/composer.json` file, then
+add it so it looks something like this
+
+```json
+    "autoload" : {
+        "psr-4" : {
+            "Bolt\\Extension\\Myname\\MyExtensionNamespace\\": "local/myname/myextension"
+        }
+    }
+```
+
+You need to make one entry for each of your local extensions. The setup here is
+identical to how you would setup your own application and there's absolutely no
+need to keep your extensions inside `extensions/local` any more. You may well
+decide to move extensions to be part of your main App namespace and if so you
+can do something like the following: 
+
+```json
+    "autoload" : {
+        "psr-4" : {
+            "MyCompany\\MyApp\\": "src"
+        }
+    }
+```
+
+With the above setup you can move local extensions to `src/Extension/` and
+access the class via `new MyCompany\MyApp\Extension\SuperCoolExtension()`
+
+### Activation
+
+Once your extensions can be autoloaded then one more step is needed to direct
+Bolt to load them.
+
+If you have a `.bolt.yml` file at the root of your project you can define which
+extension classes are loaded within that file. If you open the file as it is now
+then it will look something like this:
+
+```yaml
+
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+```
+
+You will need to edit that file to list which extensions are loaded, note that
+it is the main  extension class that needs to be specified. Assuming the
+autoload setup discussed in the last section your new `.bolt.yml` will now look
+like:
+
+```yaml
+
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+
+extensions:
+    - Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass
+```
+
+To clarify, the value you put in the yml file is exactly what you would use to
+instantiate the  class, so in code the above is equivalent to 
+`new Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass()`
+
+If you used the other autoload structure discussed, your file would look like
+this:
+
+```yaml
+paths:
+    web: public
+    themebase: public/theme
+    files: public/files
+    view: public/bolt-public/view
+
+extensions:
+    - MyCompany\MyApp\Extension\SuperCoolExtension
+```
+
+If you use a php file to bootstrap your Bolt application then you will also need
+to add each extension using php code. Your bootstrap file is at `.bolt.php` and
+it currently probably looks like this:
+
+```php
+$configuration = new MyApp\Configuration(__DIR__);
+$app = new Bolt\Application(array('resources'=>$configuration));
+// Any other application setup carries on here…
+
+// Adding your extension to Bolt
+$app['extensions']->add(new Bolt\Extension\Myname\MyExtensionNamespace\ExtensionClass());
+
+$app->initialize();
+
+return $app;
+```
```

Comment:
bear in mind that this is to help people migrating from 3.2 to 3.3, PathResolver hasn't even been included in one release yet, so there's no point documenting it in the migration guide since no-one will be using the code in production.

there's definitely quite a lot of people using a custom bootstrap in production, so it makes sense to cover it off in the migration guide. Once everyone has moved onto the new system then this page can be phased out, it's not intended to stick around forever.